src/overmind/searchFilters/effects.ts

```diff
@@ -0,0 +1,27 @@
+import axios from 'axios'
+
+import { EffectResponse } from '@/@types/EffectResponse'
+import { GovernanceResponseData } from '@/@types/State'
+import { defaultDomain, defaultHeaders } from '@/utils/RequestHelpers'
+
+export const getGovernanceData = async (): Promise<EffectResponse<GovernanceResponseData>> => {
+  return new Promise((resolve) => {
+    let result: EffectResponse<GovernanceResponseData>
+    axios
+      .get(`${defaultDomain}/governance/map`, defaultHeaders)
+      .then((body) => {
+        console.log(body.data)
+        if (body.data.lastUpdated && body.data.governanceData) {
+          const governanceResponseData = body.data as GovernanceResponseData
+          result = { data: governanceResponseData }
+        } else {
+          result = { error: { code: body.status.toString(), message: body.statusText } }
+        }
+        resolve(result)
+      })
+      .catch(() => {
+        result = { error: { code: `400`, message: `could not fetch data` } }
+        resolve(result)
```

Comment:
Are you sure you want to resolve on error instead of reject?

Reply:
This has been the pattern that was established by Mederic previously. The EffectResponse contains both a generic data perimeter and an ErrorResponse.

Reply:
@crimson-med  and @MingLiang-PALOIT this is not the way this kind of pattern should be written. You should rely on promise lifecyle:

1. before starting the job, init an empty response object
2. within `then` set the data
3. within `catch` set the error
4. within `finally` resolve your promise